### ISSUE TITLE
Button Dimension Inconsistencies

### ISSUE THREAD

**Issue by iheartrachie** [mxenabled/mx-react-components#239]
If a button only has an icon, its height is shorter by 6px than a button with text.
<img width="248" alt="screen shot 2016-03-29 at 6 35 05 pm" src="https://cloud.githubusercontent.com/assets/1916697/14128316/08af5024-f5dd-11e5-951c-280367d4e800.png">
The dimension of buttons is set by padding and margins and thus why the height is different. I was able to work around this when creating the loading button but it's causing problems when we mix icon-only buttons and buttons with text in buttonGroups.
<img width="343" alt="screen shot 2016-03-29 at 6 08 42 pm" src="https://cloud.githubusercontent.com/assets/1916697/14128293/dc0a7c10-f5dc-11e5-94bf-bb12b2e722a2.png">

If I try to work around this by setting the `minHeight: 15` on the button, the icon is offset. I can't override the negative margins used on the icon because it's set on the `Icon` itself.
<img width="329" alt="screen shot 2016-03-29 at 6 32 24 pm" src="https://cloud.githubusercontent.com/assets/1916697/14128268/a26d1256-f5dc-11e5-87f6-75e5ab1ddbf9.png">

A better way to handle this at the button level rather than trying to make up for it in the respective pieces.

**Comment by iheartrachie** [comment]
@derek-boman, @bsbeeks I also noticed that we're setting `minWidth: 50` but icon-only buttons in the style guide are much smaller, about 40px wide. Would I be okay to remove that width minimum or do we need it for something?

**Comment by iheartrachie** [comment]
To allow the spinner enough room to rotate on the loading button I had to set a minWidth afterall. I knocked it down to 16px which made the button width 42px wide overall. It's as small as I could go but should look much better in the button group. I'll work on getting a PR up for this now.
![afzh6odrsx](https://cloud.githubusercontent.com/assets/1916697/14160801/1f25beb4-f69c-11e5-9a10-14b02f244da1.gif)
